packages/react-router/src/router.ts

```diff
@@ -1545,6 +1543,7 @@ export class Router<
           location: next,
           checkLatest: () => this.checkLatest(promise),
           onReady: async () => {
+            // eslint-disable-next-line ts/require-await
```

Comment:
since we are awaiting this, shouldn't `this.startViewTransition` then return a promise?

Reply:
eslint complains that there is no await 🤷

Reply:
if I remove the `await` and this disable comment, eslint does not complain

Reply:
Let’s do that then